### DISCUSSION TITLE
Restore catmull sliders to the editor slider control type selection

### DISCUSSION THREAD

**Discussion by NiceAesth** [ppy/osu#26516] [Ideas] [answered]
Opening this as a discussion, but I noticed that the ability to select catmull sliders was replaced with B-Splines in the PR which added hand-drawn sliders.

This feels as a very odd choice as catmull would have easily still fit in the context menu and b-splines are functionally identical to bezier sliders for anything with less than 5 anchors. Having catmulls back in the editor was a considerable "cool" factor for mappers who wished to pick up the new editor as it no longer required notepad editing.

It may also cause confusion as existing catmull sliders will have the option present when right clicking a node, but if the type is changed there is no way to get it back.

**Comment by peppy** [comment, marked as answer]
This was a game direction choice by me. Catmull are fugly and I don't want them accessible.

**Reply by Empika1** [reply]
Can you please reconsider this? Catmull sliders allow for some shapes that can't be easily made by any other means, and can look quite nice in certain situations. I don't like having to copypaste an old slider every time I want a catmull.

**Reply by peppy** [reply]
Please enlighten. If we can't make the shapes better with other tools I'll reconsider

**Comment by ziv-vy** [comment]
Maybe we could keep catmull exclusively behind the keyboard shortcut `alt-5` as it is not really used and I agree that it clutters the context menu, but hey @peppy you (or someone else in your team) already coded it - why not let users have it as a shortcut? They know what they're doing if they used a keybind.

Also this would be a great opportunity to hide B-Spline too as again it is barely used.

Regarding use cases, catmull can be used when you want (somewhat) sharp sliders but linear would be too ugly. For example, I wanted to make a beginner diff on [this](https://youtu.be/Xx3QSxV4ZKQ?si=DUZx2fUKM3iGWBTN&t=63) segment of the song. Because it is beginner, I have to use sliders there. Also, because both SV and CS are very low sharp sliders would look ugly
![image](https://github.com/user-attachments/assets/b94a8fd9-2c20-4ed8-839d-c8af19b0c76f)

Sure, I _could_ use bezier, but I still want something (relatively) sharp. Bezier kinda feels like a compromise for me. 
![image](https://github.com/user-attachments/assets/0338a004-10a5-4820-894a-69830f28abd4)

Catmull would work best in this scenario IMO.

Yes, this scenarios are rare - that's why I am proposing a keybind, not a new context menu option.

**Reply by peppy** [reply]
Please provide an .osu or anything really showing the shape you're trying to make and I'll show you why catmull is not the answer.

I can GUARANTEE it is NOT what you're after.
